src/plone/restapi/serializer/dxcontent.py

```diff
@@ -220,23 +220,24 @@ def __init__(self, context, request):
 
     def __call__(self):
         primary_field_name = self.get_primary_field_name()
+        if not primary_field_name:
+            return
         for schema in iterSchemata(self.context):
             read_permissions = mergedTaggedValueDict(schema, READ_PERMISSIONS_KEY)
 
             for name, field in getFields(schema).items():
```

Comment:
Would:
```
field = getFields(schema).get(primary_field_name)
```
work, rather than looping on all the schema fields?

Reply:
I did not want to change too much, but testing it seems to work, so yes, let's go for this.  Change made.